src/plugins/links/public/embeddable/links_embeddable.tsx

```diff
@@ -6,152 +6,160 @@
-      showSaveModal: true,
-      saveModalTitle: this.getTitle(),
-    });
-  }
-
-  public async getInputAsValueType(): Promise<LinksByValueInput> {
-    return this.attributeService.getInputAsValueType(this.getExplicitInput());
-  }
-
-  public async reload() {
-    if (this.isDestroyed) return;
-    // By-reference embeddable panels are reloaded when changed, so update the attributes
-    this.initializeSavedLinks();
-    if (this.domNode) {
-      this.render(this.domNode);
-    }
-  }
-
-  public destroy() {
-    this.isDestroyed = true;
-    super.destroy();
-    this.subscriptions.unsubscribe();
-    if (this.domNode) {
-      unmountComponentAtNode(this.domNode);
-    }
-  }
-
-  public render(domNode: HTMLElement) {
-    this.domNode = domNode;
-    if (this.isDestroyed) return;
-    super.render(domNode);
-
-    this.domNode.setAttribute('data-shared-item', '');
-
-    return (
-      <LinksContext.Provider value={this}>
-        <LinksComponent />
-      </LinksContext.Provider>
-    );
-  }
-}
+  CONTENT_ID,
+  DASHBOARD_LINK_TYPE,
+  LinksAttributes,
+  LINKS_HORIZONTAL_LAYOUT,
+  LINKS_VERTICAL_LAYOUT,
+} from '../../common/content_management';
+import { DashboardLinkComponent } from '../components/dashboard_link/dashboard_link_component';
+import { ExternalLinkComponent } from '../components/external_link/external_link_component';
+import { LinksApi, LinksSerializedState } from './types';
+import { APP_NAME } from '../../common';
+import { intializeLibraryTransforms } from './initialize_library_transforms';
+import { initializeLinks } from './initialize_links';
+import { extractReferences, injectReferences } from '../../common/persistable_state';
+
+import '../components/links_component.scss';
+
+export const LinksContext = createContext<LinksApi | null>(null);
+
+export const getLinksEmbeddableFactory = () => {
+  const linksEmbeddableFactory: ReactEmbeddableFactory<LinksSerializedState, LinksApi> = {
+    type: CONTENT_ID,
+    deserializeState: (state) => {
+      const serializedState = cloneDeep(state.rawState) as LinksSerializedState;
+      if (serializedState === undefined) return {};
+
+      // by-reference embeddable
+      if (!('attributes' in serializedState) || serializedState.attributes === undefined) {
+        return serializedState;
+      }
+
+      const { attributes: attributesWithInjectedIds } = injectReferences({
+        attributes: serializedState.attributes as unknown as LinksAttributes,
+        references: state.references ?? [],
+      });
+
+      return {
+        ...serializedState,
+        attributes: attributesWithInjectedIds,
+      };
+    },
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+      const { linksApi, linksComparators, serializeLinks } = await initializeLinks(
+        state,
+        uuid,
+        parentApi
+      );
+
```

Comment:
Because we are only initializing the links when we build the embeddable, we aren't accounting for title changes like so:


https://github.com/elastic/kibana/assets/8698078/391dc4be-6687-4e3c-b924-6a7d7951635e

This is a regression from the old behaviour, which used to update the title of the current dashboard link to match the title of the current dashboard :)

Reply:
Oh, duh 🙈 I saw the `panelTitle` and just assumed it was for... panels. LOL. Makes sense that the Dashboard container is also handling this. Maybe at some point this could just be `title`? 

Thanks so much for fixing this!! 🎉